app/models/katello/root_repository.rb

```diff
@@ -322,6 +333,13 @@ def ensure_valid_retain_package_versions_count
       end
     end
 
+    def ensure_metadata_expire_valid
+      return if self.metadata_expire.blank?
+      unless yum? && custom?
+        errors.add(:metadata_expire, N_("is only allowed for Non-Redhat Yum repositories."))
```

Comment:
`errors.add(:metadata_expire, N_("Is only allowed for Non-Redhat Yum repositories."))`